### PR TITLE
fix: #5017 Open api formats to be compliant with json schema

### PR DESCRIPTION
**Pull request by christophd**
If we generate json schemas from Open API specifications in API provider we have to make sure to not use unsupported data formats (e.g. int64, int32, byte, float, etc.) in those schemas. This will lead to parsing errors when the schema is read.

This is because Open API introduces data formats that are not supported in Json Schema.

I have added a sanitize step for schemas generated from Open API specifications to not include the unsupported data format information.

### CODE REVIEW

File: app/server/api-generator/src/main/java/io/syndesis/server/api/generator/swagger/util/JsonSchemaHelper.java [hunk 1]
```diff
@@ -127,6 +132,34 @@ public static String serializeJson(final JsonNode schemaNode) {
...
+
+        if (node.isArray()) {
+            ArrayNode arrayNode = (ArrayNode) node;
+            Iterator<JsonNode> elements = arrayNode.elements();
+            elements.forEachRemaining(JsonSchemaHelper::sanitize);
+        } else if (node.isObject()) {
+            ObjectNode objectNode = (ObjectNode) node;
+
+            JsonNode formatNode = node.get("format");
+            if (formatNode != null &&
```
Comment (zregvart) [inline app/server/api-generator/src/main/java/io/syndesis/server/api/generator/swagger/util/JsonSchemaHelper.java]: Perhaps using that `isKnownFormat` helper method.

File: app/server/api-generator/src/main/java/io/syndesis/server/api/generator/swagger/util/JsonSchemaHelper.java [hunk 2]
```diff
@@ -127,6 +132,34 @@ public static String serializeJson(final JsonNode schemaNode) {
         }
     }
 
+    public static JsonNode sanitize(final JsonNode node) {
+        if (node == null) {
+            return node;
```
[the remark this answers is not in the document]
Comment (christophd) [reply app/server/api-generator/src/main/java/io/syndesis/server/api/generator/swagger/util/JsonSchemaHelper.java]: Agreed!

File: app/server/api-generator/src/main/java/io/syndesis/server/api/generator/swagger/util/JsonSchemaHelper.java (hunk 1 above)
Comment (christophd) [reply app/server/api-generator/src/main/java/io/syndesis/server/api/generator/swagger/util/JsonSchemaHelper.java]: Thanks, good idea!

**Comment by christophd** [comment]
@zregvart as usual good ideas while nitpicking, many thanks! pushed changes to this PR

**Pull request merged**